apps/password-hasher/src/lib.rs

```diff
@@ -0,0 +1,66 @@
+use argon2::{
+    password_hash::{rand_core::OsRng, PasswordHash, PasswordHasher, PasswordVerifier, SaltString},
+    Argon2, Params,
+};
+
+use spin_sdk::http::{IntoResponse, Request, Response};
+use spin_sdk::http_component;
+
+#[http_component]
+fn handle_request(req: Request) -> anyhow::Result<impl IntoResponse> {
+    let mut password: String = "boson42".to_string();
+    let mut iterations: u32 = 5;
+    let mut mem_kib: u32 = 1000;
+
+    let parsed_url = url::Url::parse(req.uri()).unwrap();
+    let hash_query: std::collections::HashMap<_, _> =
+        parsed_url.query_pairs().into_owned().collect();
+
+    for (key, value) in hash_query.into_iter() {
+        match &*key {
+            // sleep=<ms> param simulates generic i/o time
+            "sleep" => {
+                let ms = value.parse().expect("invalid sleep");
+                std::thread::sleep(std::time::Duration::from_millis(ms));
+            }
+            // password=<password> param is the user-supplied password
+            "password" => {
+                password = value.parse::<String>().expect("invalid password string");
+            }
+            // cpu=<iterations> param increases compute time
+            "cpu" => {
+                iterations = value.parse().expect("invalid cpu iteration count");
+            }
+            // mem=<KiB> param increases memory usage
+            "mem" => {
```

Comment:
In its current form, CPU usage will still be heavy when mem usage is notched up, so this isn't a good method for a "mem-intensive-only".  We do have https://github.com/fermyon/spinkube-performance/tree/main/apps/mem-intensive or we can delete that one and update this app to try to isolate memory-bound behavior better.